### DISCUSSION TITLE
Question for the Router Detach during Openthread operation

### DISCUSSION THREAD

**Discussion by eric-bj-kang** [openthread/openthread#7910] [Q&A] [answered]
Hi Team,

I'm using Openthread v1.1.
In the some cases, some node's role was change from Router to Child.

[Q.1] Would you let me know why/when this phenomenon is occurred?
[Q.2] Would you let me know how to find the reason in the source?
      - Please tell me the check point to find the root cause?

Actually Router #3 was changed to Child #1 during the Thread operation.
![image](https://user-images.githubusercontent.com/108657432/179434238-c80effe5-2fd5-4cda-8c2e-f4d39dc1bf39.png)

Best Regards,
Eric.

**Comment by jwhui** [comment, marked as answer]
Router capable devices always attach first as a child then upgrade to a router. If an active router loses connectivity with the current partition, it will attempt to reattach. A router determines that it has lost connectivity when it no longer receives sequence number updates from the Leader. The sequence number updates are propagated in MLE Advertisement messages via Leader Data TLVs.

Relevant code is here: https://github.com/openthread/openthread/blob/e538aad0951c87cfdcaf678a5c64a0acc53b5b58/src/core/thread/mle_router.cpp#L1917-L1921

**Comment by eric-bj-kang** [comment]
I understood the Router Detach occurred when a router didn't receive sequence number updates from the Leader.
[Q.3] Is that the only cause of the Router Detach?

If right, would you let me know why a router didn't receive sequence number updates?
I should find the root cause for the Router Detach.
[Q.4] Please let me know the method how to find/debug the root cause.

[Q.5] Additionally would you let me know where I can find the relative information/detailed description for the role change like as below?
- Router capable devices always attach first as a child then upgrade to a router.

**Reply by jwhui** [reply]
> [Q.3] Is that the only cause of the Router Detach?

A Router can also detach if it sees a better partition to migrate to. But in the scenario you described, there is only one partition.

> [Q.4] Please let me know the method how to find/debug the root cause.

[Enabling logs](https://openthread.io/guides/build/logs) and increasing the log level will help provide more information about why a device is detaching.

> [Q.5] Additionally would you let me know where I can find the relative information/detailed description for the role change like as below?

See the [Thread Protocol Specification](https://www.threadgroup.org/support#specifications).
